bdk_electrum_example/src/main.rs

```diff
@@ -0,0 +1,201 @@
+mod electrum;
+
+use std::fmt::Debug;
+
+use bdk_core::{bitcoin::Transaction, sparse_chain::SparseChain, BlockId, TxHeight};
+use bdk_keychain::{KeychainScan, KeychainTracker};
+use electrum::ElectrumClient;
+
+use bdk_cli::{
+    anyhow::{self, Context, Result},
+    clap::{self, Subcommand},
+};
+use log::debug;
+
+use electrum_client::ElectrumApi;
+
+#[derive(Subcommand, Debug, Clone)]
+enum ElectrumCommands {
+    /// Scans the addresses in the wallet using esplora API.
+    Scan {
+        /// When a gap this large has been found for a keychain it will stop.
+        #[clap(long, default_value = "5")]
+        stop_gap: usize,
+    },
+    /// Scans particular addresses using esplora API
+    Sync {
+        /// Scan all the unused addresses
+        #[clap(long)]
+        unused: bool,
+        /// Scan the script addresses that have unspent outputs
+        #[clap(long)]
+        unspent: bool,
+        /// Scan every address that you have derived
+        #[clap(long)]
+        all: bool,
+    },
+}
+
+fn fetch_transactions<K: Debug + Ord + Clone>(
+    new_sparsechain: &SparseChain<TxHeight>,
+    client: &ElectrumClient,
+    tracker: &KeychainTracker<K, TxHeight>,
+) -> Result<Vec<(Transaction, Option<TxHeight>)>> {
+    // Changeset of txids, both new and old.
+    let txid_changeset = tracker.chain().determine_changeset(new_sparsechain)?.txids;
+
+    // Only filter for txids that are new to us.
+    let new_txids = txid_changeset
+        .iter()
+        .filter_map(|(txid, index)| {
+            if !tracker.graph().contains_txid(*txid) {
+                Some((txid, index))
+            } else {
+                None
+            }
+        })
+        .collect::<Vec<_>>();
+
+    // Remaining of the transactions that only changed in Index
+    let existing_txs = txid_changeset
+        .iter()
+        .filter_map(|(txid, index)| match tracker.graph().tx(*txid) {
+            Some(tx) => Some((tx.clone(), *index)),
+            // We don't keep the index for `TxNode::Partial`s
+            _ => None,
+        })
+        .collect::<Vec<_>>();
+
+    let new_transactions = client.batch_transaction_get(new_txids.iter().map(|(txid, _)| *txid))?;
+
+    // Add all the transaction, new and existing into scan_update
+    let transaction_update = new_transactions
+        .into_iter()
+        .zip(new_txids.into_iter().map(|(_, index)| *index))
+        .chain(existing_txs)
+        .collect::<Vec<_>>();
+
+    Ok(transaction_update)
+}
+
+fn main() -> anyhow::Result<()> {
+    let (args, keymap, mut tracker, mut db) = bdk_cli::init::<ElectrumCommands, _>()?;
+
+    let client = ElectrumClient::new("ssl://electrum.blockstream.info:60002")?;
```

Comment:
needs to be different for different networks?

Reply:
I feel stupid but I can't seem to find any docs out there for electrum with other network ports.. Any help??